### PR TITLE
Initial Implementation

### PR DESCRIPTION
**Pull request by thypon**

### CODE REVIEW

File: assets/reviewdog/reviewdog.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        -c p/nginx \
+        -c p/docker \
+        -c p/terraform \
+        -c p/secrets \
+        $(find semgrep_rules -name '*.yml' | sed 's/^/-c /g') \
+        --metrics=off \
+        --baseline-commit origin/${GITHUB_BASE_REF:-main} \
+        --json \
+        | jq -r '.results[] | "\(.extra.severity[0:1]):\(.path):\(.end.line) \(.extra.message)"' \
+        | sed 's/$/ (Cc @brave\/sec-team @thypon @bcaller)/g' | tee semgrep.log &&\
```
Comment (thypon) [inline assets/reviewdog/reviewdog.yml]: Use the assignee variable here

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      shell: bash
+      run: |
+        python3 -m pip install --ignore-installed semgrep
+        sudo apt-get install -qy libxml2-utils
+    - name: Run reviewdog
+      shell: bash
+      env:
+        REVIEWDOG_GITHUB_API_TOKEN: ${{ inputs.github_token }}
+      run: |
+        reviewdog -reporter=github-pr-review -runners=semgrep -tee
```
Comment (thypon) [inline action.yml]: Use the config in assets/

File: assets/reviewdog/reviewdog.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      [ "$(git --no-pager diff --name-only HEAD $(git merge-base HEAD origin/${GITHUB_BASE_REF:-main}) | xargs ls  -d 2>/dev/null)" != "" ] &&\
+      semgrep \
+        -c p/ci \
+        -c p/security-audit \
+        -c p/xss \
+        -c p/nginx \
+        -c p/docker \
+        -c p/terraform \
+        -c p/secrets \
+        $(find semgrep_rules -name '*.yml' | sed 's/^/-c /g') \
```
Comment (thypon) [inline assets/reviewdog/reviewdog.yml]: Use the rules in assets

File: t3sts/semgrep_rules/nodejs_api_path_traversal.ts [hunk 4]
```diff
@@ -1 +1 @@
-fetch(`/x/${v}`, ...)
\ No newline at end of file
+await fetch(`${env.API_HOSTNAME}/api/subscriptions/${subscriptionId}`, 1337)
```
Comment (thypon) [inline t3sts/semgrep_rules/nodejs_api_path_traversal.ts]: for some reason the test does not work with the ruleset regarding fetch, can you take a look @bcaller ?

Comment (bcaller) [reply t3sts/semgrep_rules/nodejs_api_path_traversal.ts]: The custom semgrep_rules were not being loaded at all.

**Pull request merged**